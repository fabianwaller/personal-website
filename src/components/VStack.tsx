import { cn } from "@/lib/utils";

type VStackProps = {
  children: React.ReactNode;
  className?: string;
  narrow?: boolean;
};

const VStack: React.FC<VStackProps> = ({ children, className, narrow }) => (
  <div
    className={cn([
      "flex flex-col items-center justify-center gap-4",
      narrow && "gap-2",
      className,
    ])}
  >
    {children}
  </div>
);

export default VStack;
