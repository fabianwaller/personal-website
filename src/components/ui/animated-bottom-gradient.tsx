export const BottomGradient = () => {
  return (
    <>
      <span className="absolute inset-x-0 -bottom-px block h-px w-full bg-gradient-to-r from-transparent via-primary to-transparent opacity-0 transition duration-500 group-hover/btn:opacity-100" />
      <span className="absolute inset-x-10 -bottom-px mx-auto block h-px w-1/2 bg-gradient-to-r from-transparent via-primary-foreground to-transparent opacity-0 blur-sm transition duration-500 group-hover/btn:opacity-100" />
    </>
  );
};
