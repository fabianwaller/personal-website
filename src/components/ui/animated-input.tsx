import * as React from "react";
import { cn } from "@/lib/utils";
import { Input as BaseInput } from "./input";
import AnimatedRadial from "./animated-radial";

export interface InputProps
  extends React.InputHTMLAttributes<HTMLInputElement> {}

const Input = React.forwardRef<HTMLInputElement, InputProps>(
  ({ className, type, ...props }, ref) => {
    return (
      <AnimatedRadial>
        <BaseInput type={type} className={cn(className)} ref={ref} {...props} />
      </AnimatedRadial>
    );
  },
);
Input.displayName = "Input";

export { Input };
