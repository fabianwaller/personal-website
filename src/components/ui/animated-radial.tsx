import * as React from "react";
import { useMotionTemplate, useMotionValue, motion } from "framer-motion";

const AnimatedRadial = ({ children }: { children: React.ReactNode }) => {
  const radius = 600;
  const [visible, setVisible] = React.useState(false);

  let mouseX = useMotionValue(0);
  let mouseY = useMotionValue(0);

  function handleMouseMove({ currentTarget, clientX, clientY }: any) {
    let { left, top } = currentTarget.getBoundingClientRect();

    mouseX.set(clientX - left);
    mouseY.set(clientY - top);
  }
  return (
    <motion.div
      style={{
        background: useMotionTemplate`
        radial-gradient(
          ${visible ? radius + "px" : "0px"} circle at ${mouseX}px ${mouseY}px,
             hsl(var(--primary)),
          transparent 80%
        )
      `,
      }}
      onMouseMove={handleMouseMove}
      onMouseEnter={() => setVisible(true)}
      onMouseLeave={() => setVisible(false)}
      className="group/input rounded-lg p-[2px] transition duration-300"
    >
      {children}
    </motion.div>
  );
};

export default AnimatedRadial;
