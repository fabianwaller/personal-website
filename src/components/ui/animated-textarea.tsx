import * as React from "react";
import { cn } from "@/lib/utils";
import { Textarea as BaseTextArea } from "./textarea";
import AnimatedRadial from "./animated-radial";

export interface InputProps
  extends React.InputHTMLAttributes<HTMLTextAreaElement> {}

const Textarea = React.forwardRef<HTMLTextAreaElement, InputProps>(
  ({ className, type, ...props }, ref) => {
    return (
      <AnimatedRadial>
        <BaseTextArea className={cn(className)} ref={ref} {...props} />
      </AnimatedRadial>
    );
  },
);
Textarea.displayName = "Textarea";

export { Textarea };
