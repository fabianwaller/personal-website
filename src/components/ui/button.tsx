import * as React from "react";
import { Slot } from "@radix-ui/react-slot";
import { cva, type VariantProps } from "class-variance-authority";

import { cn } from "@/lib/utils";

const buttonVariants = cva(
  "inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50",
  {
    variants: {
      size: {
        default: "h-10 px-4 py-2",
        sm: "h-9 rounded-md px-3",
        lg: "h-11 rounded-md px-8",
        icon: "h-10 w-10",
      },
      variant: {
        default: "bg-primary text-title-normal hover:bg-primary-foreground",
        destructive:
          "bg-destructive text-destructive-foreground hover:bg-destructive/90",
        outline:
          "border border-input bg-background hover:bg-accent hover:text-accent-foreground",
        secondary:
          "bg-secondary text-secondary-foreground hover:bg-secondary/80",
        ghost: "hover:bg-card-foreground hover:text-primary",
        link: "text-md h-auto p-2 text-primary underline-offset-4 hover:underline",
        unstyled: "p-0 text-primary hover:text-accent-foreground",
        black:
          "group/btn relative block h-10 w-full rounded-md bg-gradient-to-br from-black to-neutral-600 font-medium text-white shadow-[0px_1px_0px_0px_#ffffff40_inset,0px_-1px_0px_0px_#ffffff40_inset] dark:bg-zinc-800 dark:from-zinc-900 dark:to-zinc-900 dark:shadow-[0px_1px_0px_0px_var(--zinc-800)_inset,0px_-1px_0px_0px_var(--zinc-800)_inset]",
        white:
          "group/btn relative flex h-10 w-full items-center justify-start space-x-2 rounded-md bg-gray-50 px-4 font-medium text-black shadow-input dark:bg-zinc-900 dark:shadow-[0px_0px_1px_1px_var(--neutral-800)]",
      },
    },
    defaultVariants: {
      variant: "default",
      size: "default",
    },
  },
);

export interface ButtonProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement>,
    VariantProps<typeof buttonVariants> {
  asChild?: boolean;
}

const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
  ({ className, variant, size, asChild = false, ...props }, ref) => {
    const Comp = asChild ? Slot : "button";
    return (
      <Comp
        className={cn(buttonVariants({ variant, size, className }))}
        ref={ref}
        {...props}
      />
    );
  },
);
Button.displayName = "Button";

export { Button, buttonVariants };
