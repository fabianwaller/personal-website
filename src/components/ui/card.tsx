import * as React from "react";

import { cn } from "@/lib/utils";

const Card = React.forwardRef<
  HTMLDivElement,
  React.HTMLAttributes<HTMLDivElement>
>(({ className, ...props }, ref) => (
  <div
    ref={ref}
    className={cn(
      "rounded-lg border bg-card shadow-sm hover:shadow-md",
      className,
    )}
    {...props}
  />
));
Card.displayName = "Card";

const CardHeader = React.forwardRef<
  HTMLDivElement,
  React.HTMLAttributes<HTMLDivElement>
>(({ className, ...props }, ref) => (
  <div
    ref={ref}
    className={cn("flex flex-col space-y-1.5 p-6", className)}
    {...props}
  />
));
CardHeader.displayName = "CardHeader";

const CardTitle = React.forwardRef<
  HTMLParagraphElement,
  React.HTMLAttributes<HTMLHeadingElement>
>(({ className, ...props }, ref) => (
  <h3
    ref={ref}
    className={cn(
      "text-lg font-medium leading-none tracking-tight text-title-normal",
      className,
    )}
    {...props}
  />
));
CardTitle.displayName = "CardTitle";

const CardDescription = React.forwardRef<
  HTMLParagraphElement,
  React.HTMLAttributes<HTMLParagraphElement>
>(({ className, ...props }, ref) => (
  <p ref={ref} className={cn("text-normal", className)} {...props} />
));
CardDescription.displayName = "CardDescription";

const CardContent = React.forwardRef<
  HTMLDivElement,
  React.HTMLAttributes<HTMLDivElement>
>(({ className, ...props }, ref) => (
  <div ref={ref} className={cn("p-6 pt-0", className)} {...props} />
));
CardContent.displayName = "CardContent";

const CardFooter = React.forwardRef<
  HTMLDivElement,
  React.HTMLAttributes<HTMLDivElement>
>(({ className, ...props }, ref) => (
  <div
    ref={ref}
    className={cn("flex items-center p-6 pt-0", className)}
    {...props}
  />
));
CardFooter.displayName = "CardFooter";

export {
  Card,
  CardHeader,
  CardFooter,
  CardTitle,
  CardDescription,
  CardContent,
};
