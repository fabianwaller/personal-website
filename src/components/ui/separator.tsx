import * as React from "react";
import * as SeparatorPrimitive from "@radix-ui/react-separator";

import { cn } from "@/lib/utils";

const Separator = React.forwardRef<
  React.ElementRef<typeof SeparatorPrimitive.Root>,
  React.ComponentPropsWithoutRef<typeof SeparatorPrimitive.Root>
>(
  (
    { className, orientation = "horizontal", decorative = true, ...props },
    ref,
  ) => (
    <SeparatorPrimitive.Root
      ref={ref}
      decorative={decorative}
      orientation={orientation}
      className={cn(
        "shrink-0 bg-border",
        orientation === "horizontal" ? "h-[1px] w-full" : "h-full w-[1px]",
        className,
      )}
      {...props}
    />
  ),
);
Separator.displayName = SeparatorPrimitive.Root.displayName;

{
  /* <div className="my-8 h-[1px] w-full bg-gradient-to-r from-transparent via-neutral-300 to-transparent dark:via-neutral-700" /> */
}

export { Separator };
