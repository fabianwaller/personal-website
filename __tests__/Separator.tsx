import { Separator } from "@/components/ui/separator";
import VStack from "@/components/VStack";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";

describe("Separator", () => {
  it("renders horizontal Separator unchanged", () => {
    const component = render(<Separator />);
    expect(component.asFragment()).toMatchSnapshot();
  });
  it("renders vertical Separator unchanged", () => {
    const component = render(<Separator orientation="vertical" />);
    expect(component.asFragment()).toMatchSnapshot();
  });
});
