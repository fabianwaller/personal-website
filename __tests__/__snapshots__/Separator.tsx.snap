// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Separator renders Separator unchanged 1`] = `
<DocumentFragment>
  <div
    class="shrink-0 bg-border h-[1px] w-full"
    data-orientation="horizontal"
    role="none"
  />
</DocumentFragment>
`;

exports[`Separator renders horizontal Separator unchanged 1`] = `
<DocumentFragment>
  <div
    class="shrink-0 bg-border h-[1px] w-full"
    data-orientation="horizontal"
    role="none"
  />
</DocumentFragment>
`;

exports[`Separator renders vertical Separator unchanged 1`] = `
<DocumentFragment>
  <div
    class="shrink-0 bg-border h-full w-[1px]"
    data-orientation="vertical"
    role="none"
  />
</DocumentFragment>
`;
